Zend/zend_alloc.h

```diff
@@ -21,10 +21,8 @@
 #ifndef ZEND_ALLOC_H
 #define ZEND_ALLOC_H
 
-#include <stdio.h>
-
-#include "../TSRM/TSRM.h"
-#include "zend.h"
+#include "zend_portability.h" // for BEGIN_EXTERN_C
+#include "zend_types.h" // for zend_result
```

Comment:
The issue I have with such comments is that they are very fragile. As the code changes, it may use more of the things in this file. It may stop using zend_result. zend_result may be moved to another header file. But unless the file is going to go away completely, or similar major change happens, nobody would bother to update all comments of this nature. It is not realistic to expect from people who alter the code to evaluate each time whether their code was the last instance of using some particular construct introduced in some include, and whether now we need to change the comment to some other construct also introduced by that include - nobody ever is going to do that. So this may be technically true now - but in a year, we won't know if it's true, and in three years, it becomes just meaningless.

Reply:
That would be RFC vote "Is it allowed to document an #include line with a code comment?" -  https://wiki.php.net/rfc/include_cleanup - please vote!
See also https://github.com/php/php-src/pull/10472